elfo-core/src/errors.rs

```diff
@@ -1,14 +1,40 @@
+use std::fmt::Display;
+
 use derive_more::{Display, Error};
 
 #[derive(Debug, Display, Error)]
 #[non_exhaustive]
 pub enum StartError {
-    /// Configs must be valid at the start-up.
-    #[display(fmt = "invalid config")]
-    InvalidConfig,
+    #[display(fmt = "error starting entrypoints: {}", _0)]
+    EntrypointErrors(#[error(not(source))] EntrypointErrors),
+    #[display(fmt = "other: {}", _0)]
     Other(#[error(not(source))] String),
 }
 
+#[derive(Debug, Error)]
+#[non_exhaustive]
+pub struct EntrypointErrors {
+    pub errors: Vec<EntrypointError>,
+}
+
+impl Display for EntrypointErrors {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        writeln!(f, "entrypoint errors:")?;
+        for error in &self.errors {
+            writeln!(f, "- {}", error)?;
```

Comment:
This way of listing errors does not compose. If this error is a part of a top-level list, no indentation is applied. An alternative is to list everything on the same line, which is not readable.

Reply:
On the same line is ok for the default style (but using `,` or `;` as a separator). It can be altered in the future by using the alternate mode of `Display` (`#`). Anyway, users can implement any useful form for them.

Alternatively, we can inspire by https://docs.rs/eyre/latest/eyre/struct.Report.html#display-representations